src/qwircl/grid.cljs

```diff
@@ -0,0 +1,32 @@
+(ns qwircl.grid)
+
+(defn with-positions [grid positions hand]
+  (reduce #(assoc-in %1 (:coordinates %2) (hand (first (:hand %2)))) grid positions))
+
+(def transformations {:horizontal {:x [dec inc] :y [identity identity]}
+                      :vertical {:x [identity identity] :y [dec inc]}})
+(defn get-neighbors 
+  ([grid [x y] f g initial]
+   (loop [i (f x)
+          j (g y)
+          acc initial]
+     (if-let [cell (get-in grid [i j])]
+       (recur (f i) (g j) (conj acc cell))
+       acc)))
+  ([grid clicked direction]
+   (let [[f1 f2 g1 g2] (map #(get-in transformations (apply (partial conj [direction]) %))
```

Comment:
`(apply (partial conj [...]) %)` is equivalent to `(into [...] %)` (also `(apply vector ... %)`, but I think `into` is more idiomatic here)

Reply:
Good point.